Lib/unittest/mock.py

```diff
@@ -727,11 +727,10 @@ def __delattr__(self, name):
                 # not set on the instance itself
                 return
 
-        if name in self.__dict__:
-            object.__delattr__(self, name)
-
         obj = self._mock_children.get(name, _missing)
-        if obj is _deleted:
+        if name in self.__dict__:
+            super().__delattr__(name)
+        elif obj is _deleted:
             raise AttributeError(name)
         if obj is not _missing:
```

Comment:
There was a comment in the patch about this line of checking _missing being superfluous and that it could be removed. Is there a test that will break on removing this?

Reply:
No, no test fail when removing this line. This does not mean is superfluous without inspecting. I prefer to handle that question in a different PR :)